server/auth.go

```diff
@@ -78,15 +89,65 @@ func (h *Handler) LoginPost(c echo.Context) error {
 }
 
 func (h *Handler) LoginGet(c echo.Context) error {
+	sess, _ := session.Get(CookieKeySession, c)
+
 	vars := map[string]interface{}{
 		"csrf":               c.Get("csrf").(string),
 		"globus":             viper.GetBool("globus_signup"),
 		"enable_user_signup": viper.GetBool("enable_user_signup"),
+		"success":            sess.Flashes("success"),
 	}
 
+	sess.Save(c.Request(), c.Response())
+
 	return c.Render(http.StatusOK, "login.html", vars)
 }
 
+func (h *Handler) ChangePost(c echo.Context) error {
+	message := ""
+	sess, _ := session.Get(CookieKeySession, c)
+
+	uid := c.FormValue("uid")
+	current := c.FormValue("password")
+	pass := c.FormValue("new_password")
+	pass2 := c.FormValue("new_password2")
+	challenge := c.FormValue("challenge")
+
+	if pass != pass2 {
```

Comment:
We should check cookie here to ensure user authenticated and has an expired password. Otherwise return 401.
 
We also need to add more validation here. Currently, you can just visit this URL unauthenticated `/auth/change?uid=SOMEUID` and hit submit and you get a fatal error. Ideally we would re-use the same logic as in [cpw.go](https://github.com/ubccr/mokey/blob/master/server/cpw.go#L14) as it does some server side validation on the password before sending off to FreeIPA. So perhaps we break out the password validation logic into a function `func validatePassword()` which can be called by both `changePassword()` and `setPassword`?